lib/pbench/test/unit/server/query_apis/test_controllers_list.py

```diff
@@ -12,64 +12,22 @@ class TestControllersList:
     constructor and `post` service.
     """
 
-    def build_index(self, server_config, dates):
-        """
-        Build the index list for query
-
-        Args:
-            dates (iterable): list of date strings
-        """
-        idx = server_config.get("Indexing", "index_prefix") + ".v6.run-data."
-        index = "/"
-        for d in dates:
-            index += f"{idx}{d},"
-        return index
-
-    def test_non_accessible_user_data(self, client, server_config, pbench_token):
-        """
-        Test behavior when Authorization header does not have access to other user's data
-        """
-        # The pbench_token fixture logs in as user "drb"
-        # Trying to access the data belong to the user "pp"
-        response = client.post(
-            f"{server_config.rest_uri}/controllers/list",
-            headers={"Authorization": "Bearer " + pbench_token},
-            json={"user": "pp", "start": "2020-08", "end": "2020-10"},
-        )
-        assert response.status_code == HTTPStatus.FORBIDDEN
-
-    @pytest.mark.parametrize(
-        "user", ("drb", "pp"),
-    )
-    def test_accessing_user_data_with_invalid_token(
-        self, client, server_config, pbench_token, user
-    ):
-        """
-        Test behavior when Authorization header does not have access to other user's data
-        """
-        # valid token logout
-        response = client.post(
-            f"{server_config.rest_uri}/logout",
-            headers=dict(Authorization="Bearer " + pbench_token),
-        )
-        assert response.status_code == HTTPStatus.OK
-        response = client.post(
-            f"{server_config.rest_uri}/controllers/list",
-            headers={"Authorization": "Bearer " + pbench_token},
-            json={"user": user, "start": "2020-08", "end": "2020-10"},
-        )
-        assert response.status_code == HTTPStatus.FORBIDDEN
-
-    def test_missing_json_object(self, client, server_config, pbench_token):
-        """
-        Test behavior when no JSON payload is given
-        """
-        response = client.post(
-            f"{server_config.rest_uri}/controllers/list",
-            headers={"Authorization": "Bearer " + pbench_token},
-        )
-        assert response.status_code == HTTPStatus.BAD_REQUEST
-        assert response.json.get("message") == "Invalid request payload"
+    @pytest.fixture(autouse=True)
+    def _setup(self):
+        self.endpoint = "/controllers/list"
+        self.json = {"user": "drb", "start": "2020-08", "end": "2020-10"}
+        self.json_bad_date = {"user": "drb", "start": "2020-12", "end": "2020-19"}
+        self.json_error = {"user": "", "start": "2020-08", "end": "2020-08"}
+        self.empty_response_payload = {
```

Comment:
It's good that the empty response payload is under subclass-control; making the Elastic URI configurable also would be an improvement.